src/Sets/Star.jl

```diff
@@ -166,3 +167,45 @@ Return the predicate of a star.
 A polyhedral set representing the predicate of the star.
 """
 predicate(X::STAR) = set(X)
+
+function _intersection!(c, V, P::Union{HPoly, HPolygon}, H::HalfSpace)
```

Comment:
You can also add `HPolygonOpt` to the `Union`. How about we define the abbreviation `Hrep` or `HRep` or `HREP` as in
```julia
const HREP = Union{HPoly, HPolygon, HPolygonOpt}
```
and use that instead? This interface would assert `addconstraint` and `remove_redundant_constraints`.
We could do something similar for `VREP = Union{VPolytope, VPolygon}`.

Reply:
makes sense . for another issue?